Adds WNIS July 2026 post. (#1486)
apple/swift-org-website: _posts/2026-08-06-whats-new-in-swift-july-2026.md

---
layout: new-layouts/post
published: true
date: 2026-08-06 12:00
title: "What's new in Swift: July 2026 Edition"
author: [alexandersandberg, davelester]
category: "Digest"
---

Welcome to "What's new in Swift," a curated digest of releases, videos, and discussions in the Swift project and community.

> Swift is built by contributors working on far more than the language and compiler. Alex Sandberg reflects on what that kind of contribution looks like, and why there's a place for you too:

Reviewer: ```suggestion
Swift is built by contributors working on far more than the language and compiler. Alexander Sandberg reflects on what that kind of contribution looks like, and why there's a place for you too:
```

I usually spell my name out when combined with my last name 😄

> > Hi, I'm Alex. Four years ago I joined the Swift Website Workgroup to help care for the website you're likely reading this on. I was still fairly new to the Swift community then, but I wanted to get involved somehow and make a difference.

Reviewer 2: ```suggestion
> Hi, I'm Alex. Four years ago I joined the Swift Website Workgroup to help care for the website you're reading this on. I was still fairly new to the Swift community then, but I wanted to get involved somehow and make a difference.
```

I'd drop "likely" here, since they *are* reading it on this website! 😂

Author: @alexandersandberg OK with this change?

Reviewer: Yes! I was considering RSS readers here but either way is fine 😄

> 
> Contributing to Swift sounds like it should involve working on the compiler or writing and reviewing language evolution proposals, but there's a lot going on around the language itself too. What I actually spent my time doing was helping improve the site's content and navigation, taking part in the big redesign, writing release blog posts, reviewing pull requests, and answering issues and forum posts. All of it is work anyone can learn to do, and all of it ends up in front of everyone who comes to Swift.org—which made even the smallest fixes feel worth doing.
> 

> > The Website Workgroup happened to be my way in, but there are groups like it across the project—server, testing, Windows, Android, and more—each a handful of people looking after their part of the ecosystem. Whatever you're good at or curious about, there's probably a corner of Swift that could use your help. The [community page](https://www.swift.org/community/) is a good place to start looking, and the people you'll meet are some of the friendliest I've come across in open source.

Reviewer 3: ```suggestion
> The Website Workgroup happened to be my way in, but there are groups like it across the project—server, testing, Windows, Android, and more—each a handful of people looking after their part of the ecosystem. Whatever you're good at or curious about, there's probably a corner of Swift that could use your help. The [community page](/community/) is a good place to start looking, and the people you'll meet are some of the friendliest I've come across in open source.
```

> 
> After four years, [I've stepped away to focus on other things in life](https://alexandersandberg.com/articles/leaving-the-swift-website-workgroup/). Thank you to everyone I got to work with over these years! I'm leaving the workgroup, but not the community, and I'll still be around helping push Swift forward where I can.

> For more information about workgroups, the [How we work page](https://www.swift.org/community/how-we-work/) lists all workgroups. And on the [Swift forums](https://forums.swift.org) you'll often see workgroups sharing open meetings if you're looking to get involved!

Reviewer 3: ```suggestion
For more information about workgroups, the [How we work page](/community/how-we-work/) lists all workgroups. And on the [Swift forums](https://forums.swift.org) you'll often see workgroups sharing open meetings if you're looking to get involved!
```

> For more information about workgroups, the [How we work page](https://www.swift.org/community/how-we-work/) lists all workgroups. And on the [Swift forums](https://forums.swift.org) you'll often see workgroups sharing open meetings if you're looking to get involved!

Reviewer 2: Is there anything we can do to help people find the open meetings? I know the issue is that they're not all posted in the same place, but a link to the root of the forum is a bit of a dead end. I'm not sure what to suggest, here, really, but I thought it at least worth mentioning.

Author: Thanks for the feedback on this! I agree this is an important topic but unfortunately we don't have a standard way of surfacing this info. Took a look and there don't seem to be many upcoming meetings as well. But I'd like to revisit this in the future.

* Speaking of low-level Swift: [I Built a Timer for the Playdate in Swift](https://www.youtube.com/watch?v=Fs27ooAUbzk) and [I Built a 3D Renderer for the Playdate (in Swift)](https://www.youtube.com/watch?v=-uo-c9tzYnE) are a fun pair of videos on building for the handheld game console in Swift.
* Melbourne CocoaHeads has been uploading a great back catalog of past event talks to their YouTube channel, including [Christian Mitteldorf's 2023 talk on running Swift on the server](https://www.youtube.com/watch?v=V0G6hOwJPhw).

## Recent project changes

* **Google Summer of Code contributions are underway**: Ege Kaya is building a Task Registry for the Swift Concurrency runtime ([PR #90547](https://github.com/swiftlang/swift/pull/90547)), Filip Sakellariou is contributing to swift-syntax ([PR #3378](https://github.com/swiftlang/swift-syntax/pull/3378), [PR #3390](https://github.com/swiftlang/swift-syntax/pull/3390)), and Padmashree S S is working in sourcekit-lsp ([PR #2698](https://github.com/swiftlang/sourcekit-lsp/pull/2698), [PR #2729](https://github.com/swiftlang/sourcekit-lsp/pull/2729)).

> * **Task Stealers, a new primitive in Swift Concurrency's runtime, [merged this month](https://github.com/swiftlang/swift/pull/86082)** after generating the liveliest discussion of any PR in July. They change how work gets distributed across the cooperative thread pool, an under-the-hood change worth watching if you write concurrency-heavy server or app code.

Reviewer 2: Maybe "recently" instead of "this month" given it's August, but "last month" also doesn't seem right.

Author: ```suggestion
* **Task Stealers, a new primitive in Swift Concurrency's runtime, [recently merged](https://github.com/swiftlang/swift/pull/86082)** after generating the liveliest discussion of any PR in July. They change how work gets distributed across the cooperative thread pool, an under-the-hood change worth watching if you write concurrency-heavy server or app code.
```

* **SwiftPM landed a [reference implementation](https://github.com/swiftlang/swift-package-manager/pull/10255) of the package registry service**, along with [basic auth support](https://github.com/swiftlang/swift-package-manager/pull/10258), a solid starting point if you're exploring self-hosting a registry.

> * **Small but welcome fixes landed across tooling**: the VS Code extension now [fails loudly when a Swift installation is broken](https://github.com/swiftlang/vscode-swift/pull/2080) and gained a [clean build folder command](https://github.com/swiftlang/vscode-swift/pull/2280), while Swift Testing's event stream gained an ABI.Context API that [produces human-readable output](https://github.com/swiftlang/swift-testing/pull/1759) for CI tooling and dashboards.

Reviewer 2: ```suggestion
* **Small but welcome fixes landed across tooling**: the VS Code extension now [fails loudly when a Swift installation is broken](https://github.com/swiftlang/vscode-swift/pull/2080), and gained a [clean build folder command](https://github.com/swiftlang/vscode-swift/pull/2280), while Swift Testing's event stream gained an ABI.Context API that [produces human-readable output](https://github.com/swiftlang/swift-testing/pull/1759) for CI tooling and dashboards.
```

## New package releases
* Need to tokenize text fast? [swift-gigatoken](https://github.com/1amageek/swift-gigatoken) is a zero-copy tokenizer hitting up to 670 MB/s, inspired by Marcel Roed's gigatoken, with NEON/CRC32 support and compatibility across WebAssembly and Embedded Swift.
* [any-error-swift](https://github.com/MFB-Technologies-Inc/any-error-swift) takes a different approach to error handling, offering a concrete, value-semantic, `Hashable` type that erases any Swift error, no Foundation required, and works on every platform Swift runs on.

## Swift Evolution

> The Swift project adds new language features through the [Swift Evolution process](https://www.swift.org/swift-evolution/). These are some of the proposals recently accepted for a future Swift release.

Reviewer 3: ```suggestion
The Swift project adds new language features through the [Swift Evolution process](/swift-evolution/). These are some of the proposals recently accepted for a future Swift release.
```

> The Swift project adds new language features through the [Swift Evolution process](https://www.swift.org/swift-evolution/). These are some of the proposals recently accepted for a future Swift release.

Reviewer 3: We should update the text to include swift testing, as ST-0025 is a testing feature.

Author: @shahmishal could you clarify your suggestion? Would the existing paragraph intro above be sufficient or did you have something else in mind? Additionally, the ST-0025 proposal is listed below.

Reviewer 3: ```
## Swift Evolution
The Swift project adds new language features through the [Swift Evolution process](https://www.swift.org/swift-evolution/). These are some of the proposals recently accepted for a future Swift release.
```

The current text `new language features` should be expanded to include testing because the ST-0025 is not a new language feature.

Author: @shahmishal good catch, thanks! Will remove "language" to simplify it, so it simply says "new features."
